poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1185,4 +1198,303 @@ public void syncPDUID(FacilitySoftwareConfig powerIQ) {
+         if (pdus.isEmpty()) {
+            break;
+         }
+         for (Pdu pdu : pdus) {
+            Asset asset = new Asset();
+            asset = fillLocation(pdu.getParent(), racksMap, rowsMap, aislesMap, roomsMap, floorsMap,
+                  dataCentersMap);
+            asset.setAssetName(pdu.getName());
+            asset.setSerialnumber(pdu.getSerialNumber());
+            asset.setAssetSource(powerIQ.getId());
+            asset.setCategory(AssetCategory.PDU);
+            asset.setCreated(System.currentTimeMillis());
+            List<Outlet> outlets = client.getOutlets(pdu.getId());
+            List<Inlet> inlets = client.getInlets(pdu.getId());
+            ObjectMapper mapper = new ObjectMapper();
+            String outletString = null;
+            String inletString = null;
+            try {
+               outletString = generatePduOutletString(outlets);
+               inletString = generatePduInletString(inlets);
+            } catch (JsonProcessingException e) {
+               logger.info(String.format("Sync pdu metadata error",e.getCause()));
+            }
+            Map<String,String> pduInfo = generatePduRateInfoMap(pdu);
+            if(outletString != null) {
+               pduInfo.put(FlowgateConstant.PDU_OUTLETS_FROM_POWERIQ, outletString);
+            }
+            if(inletString != null) {
+               pduInfo.put(FlowgateConstant.PDU_INLETS_FROM_POWERIQ, inletString);
+            }
+            pduInfo.put(FlowgateConstant.PDU_ID_FROM_POWERIQ, String.valueOf(pdu.getId()));
+
+            String pduInfoString = null;
+            try {
+               pduInfoString = mapper.writeValueAsString(pduInfo);
+            } catch (JsonProcessingException e) {
+               logger.info(String.format("Sync pdu metadata error",e.getCause()));
+            }
+            if(pduInfoString != null) {
+               HashMap<String, String> justfacationfields = new HashMap<String, String>();
+               justfacationfields.put(FlowgateConstant.PDU, pduInfoString);
+               asset.setJustificationfields(justfacationfields);
+            }
+            pduAssets.add(asset);
+         }
+         offset += limit;
+      }
+      return pduAssets;
+   }
+
+   /**
+    * Use the Pdu_Id which is from PowerIQ as the filter option
+    * @param existedPduAssets
+    * @param pdusFromPowerIQ
+    * @return
+    */
+   public List<Asset> getPduAssetsNeedtoSave(Map<String, Asset> existedPduAssets, List<Asset> pdusFromPowerIQ){
+      List<Asset> pdusNeedToSave = new ArrayList<Asset>();
+      if(existedPduAssets.isEmpty()) {
+         return pdusFromPowerIQ;
+      }
+      if(pdusFromPowerIQ.isEmpty()) {
+         return pdusNeedToSave;
+      }
+      ObjectMapper mapper = new ObjectMapper();
+      for(Asset asset : pdusFromPowerIQ) {
+         HashMap<String, String> justficationfields = asset.getJustificationfields();
+         if(justficationfields == null) {
```

Comment:
I guess this check is useless, since all the items in pdusFromPowerIQ is a new Asset right?

Reply:
I guess those unwrapper logic doesn't have any meaning since it's always empty.